compendium/DeclarativeServices/src/manager/states/CCActiveState.cpp

```diff
@@ -37,7 +38,12 @@ std::shared_ptr<ComponentInstance> CCActiveState::Activate(ComponentConfiguratio
   auto logger = mgr.GetLogger();
   if(latch.CountUp())
   {
-    instance = mgr.CreateAndActivateComponentInstance(clientBundle);
+    try {
```

Comment:
consider using a scope guard here to CountDown().

Reply:
implemented a scope guard class to invoke CountDown()